### PR TITLE
Add userEditPerms to manage user's plotinformation edit permissions

### LINKED ISSUE #328: Add permissions logic for Plot Information edit
When a user goes to edit the alt text `Plot Information` via the edit button, there should be a check to determine if they can edit this or not.

If they cannot edit, and they are logged in, they should not be allowed to edit at all. This could mean disabling the button or notifying the user on click.

If they are not logged in, ask the user to login.

### PR DESCRIPTION
**Pull request by JakeWags**
### Does this PR close any open issues?
Closes #328 

### Give a longer description of what this PR addresses and why it's needed
This PR removes the edit icon and "add plot information" button for users which do not have "owner" or "maintainer" permissions. Additionally, the save button callback and edit text callback are disabled for those without permissions, as a fail-safe.

### Provide pictures/videos of the behavior before and after these changes (optional)
No major visual changes (just hiding the buttons)

### Have you added or updated relevant tests?
- [X] Yes
- [ ] No changes are needed

### Have you added or updated relevant documentation?
- [X] Yes
- [ ] No changes are needed

### Are there any additional TODOs before this PR is ready to go?
None

### CODE REVIEW

**Review by JackWilb — APPROVED** [review]

File: packages/upset/src/components/Upset.tsx [hunk 1]
```diff
@@ -20,12 +20,13 @@ const defaultVisibleSets = 6;
...
  * @param {SidebarProps} [provVis] - The provenance visualization sidebar options.
  * @param {SidebarProps} [elementSidebar] - The element sidebar options. This sidebar is used for element queries, element selection datatable, and supplimental plot generation.
  * @param {SidebarProps} [altTextSidebar] - The alternative text sidebar options. This sidebar is used to display the generated text descriptions for an Upset 2.0 plot, given that the `generateAltText` function is provided.
  * @param {() => Promise<AltText>} [generateAltText] - The function to generate alternative text.
- * @returns {JSX.Element} The rendered Upset component.
+* @returns {JSX.Element} The rendered Upset component.
```
[the remark this answers is not in the document]
Comment (JakeWags) [reply packages/upset/src/components/Upset.tsx]: Addressed in https://github.com/visdesignlab/upset2/pull/415/commits/40790662251f63631b84648efbd588c4c62cc25b

File: packages/app/src/components/Body.tsx [hunk 2]
```diff
@@ -51,6 +52,23 @@ export const Body = ({ data, config }: Props) => {
...
+        const r = await api.getCurrentUserWorkspacePermissions(workspace || '');
+        // https://api.multinet.app/swagger/?format=openapi#/definitions/PermissionsReturn for possible permissions returns
+        return r.permission_label === 'owner' || r.permission_label === 'maintainer';
+      } catch (e) {
+        return false;
+      }
+    };
+
+    fetchPermissions().then(setPermissions);
+  }, [workspace]);
```
[the remark this answers is not in the document]
Comment (JakeWags) [reply packages/app/src/components/Body.tsx]: Addressed in https://github.com/visdesignlab/upset2/pull/415/commits/40790662251f63631b84648efbd588c4c62cc25b

File: README.md [hunk 3]
```diff
@@ -204,6 +204,7 @@ const main = () => {
 - `visualizeAttributes` (optional)(`string[]`): List of attribute names (strings) which should be visualized. Defaults to the first 3 if no value is provided. If an empty list is provided, displays no attributes.
 - `visualizeUpsetAttributes` (optional)(`boolean`): Whether or not to visualize UpSet generated attributes (`degree` and `deviation`). Defaults to `false`.
 - `allowAttributeRemoval` (optional)(`boolean`): Whether or not to allow the user to remove attribute columns. This should be enabled only if there is an option within the parent application which allows for attributes to be added after removal. Default attribute removal behavior in UpSet 2.0 is done via context menu on attribute headers. Defaults to `false`.
+- `userEditPerms` (optional)(`boolean`): Whether or not the user has plot information edit permissions.
```
Comment (NateLanza) [inline README.md:207]: I recommend `userCanEdit` or even `userCanEditPlotInfo` instead, but this is fine if you prefer. And I think `Whether or not the user can edit the plot information in the text descriptions sidebar` is a clearer description, but again, this is personal preference.

**Review by NateLanza — APPROVED** [review]
Param name & documentation could be clarified, otherwise LGTM.

File: README.md (hunk 3 above)
Comment (JakeWags) [reply README.md]: Resolved in https://github.com/visdesignlab/upset2/pull/415/commits/9ecb9eb6f1347bc338aa81b224dcddee605c048f

I agree with your recommendations here.

**Pull request merged**